lib/GeometryMath.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+var Cesium = require('cesium');
+var Cartesian3 = Cesium.Cartesian3;
+
+module.exports = {
+    sumBarycentric: sumBarycentric,
+    flattenTriangle: flattenTriangle,
+    pointLineDistanceParametric: pointLineDistanceParametric
+};
+
+// Functions for working with geometry like triangles and lines
+
+var sumBarycentricScratch = new Cartesian3();
+function sumBarycentric(barycentric, vector0, vector1, vector2, result) {
```

Comment:
I still want to leave the pragma because we are doing more and more heavy math here and elsewhere and we will ultimately want to optimize them out.